test/extended/templates/templateinstance_impersonation.go

```diff
@@ -21,18 +21,20 @@ import (
 // 1. Check that users can't create or update templateinstances unless they are,
 // or can impersonate, the requester.
 // 2. Check that templateinstancespecs, particularly including
-// requester.username, are immutable.
+// requester.username and groups, are immutable.
 var _ = g.Describe("[templates] templateinstance impersonation tests", func() {
 	defer g.GinkgoRecover()
 
 	var (
 		cli = exutil.NewCLI("templates", exutil.KubeConfigPath())
 
-		adminuser       *userapi.User // project admin, but can't impersonate anyone
-		impersonateuser *userapi.User // project edit, and can impersonate edituser1
-		edituser1       *userapi.User // project edit, can be impersonated by impersonateuser
-		edituser2       *userapi.User // project edit
-		viewuser        *userapi.User // project view
+		adminuser              *userapi.User // project admin, but can't impersonate anyone
+		impersonateuser        *userapi.User // project edit, and can impersonate edituser1
+		impersonatebygroupuser *userapi.User
```

Comment:
I'm not sure what the basis is for adding impersonatebygroupuser and impersonategroup - can you talk me through it?